it/src/main/java/com/google/cloud/teleport/it/PerformanceBenchmarkingBase.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it;
+
+import static com.google.cloud.teleport.it.logging.LogStrings.formatForLogging;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.auth.Credentials;
+import com.google.cloud.bigquery.InsertAllRequest.RowToInsert;
+import com.google.cloud.teleport.it.bigquery.BigQueryResourceManager;
+import com.google.cloud.teleport.it.bigquery.DefaultBigQueryResourceManager;
+import com.google.cloud.teleport.it.dataflow.DataflowClient;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.JobInfo;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator;
+import com.google.cloud.teleport.it.dataflow.DefaultDataflowClient;
+import com.google.cloud.teleport.it.monitoring.MonitoringClient;
+import com.google.common.base.MoreObjects;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Integration test for testing performance of dataflow templates. */
+@RunWith(JUnit4.class)
+public class PerformanceBenchmarkingBase {
```

Comment:
Something to consider is for BenchmarkingBase to extend TemplateTestBase, so you get all the template staging for free (and DataflowClient as well).

Reply:
That is a good idea. I didn't do that since PerformanceBenchmarkingBase will be used to test regular dataflow pipelines as well. 
We can extend the integration test class from both PerformanceBenchmarkingBase and TemplateTestBase to get everything.